### PR TITLE
add orchestrator to cfn

### PR DESCRIPTION
**Pull request by HieronymusLex**
* Adds orchestrator function to CFN
* Moves libs to Lambda Layers
* Moves Setup Lambda role definition to intrinsic role by SAM

### CODE REVIEW

File: src/cfn/template.yaml [hunk 1]
```diff
@@ -453,6 +435,57 @@ Resources:
...
+        - nodejs8.10
+        - nodejs10.x
+
+  Orchestrator:
+    Type: AWS::Serverless::Function
+    Properties:
+      Handler: index.handler
+      Runtime: nodejs10.x
+      CodeUri: ../backend/functions/orchestrator/dist/
+      Timeout: 10
```
Comment (matteofigus) [inline src/cfn/template.yaml]: Is 10 enough? I am thinking of the scenario of Fargate stopping the container by sending the SIG_TERM and allowing the app to gracefully save to dynamo, finish up and all the things.

File: src/cfn/template.yaml [hunk 2]
```diff
@@ -453,6 +435,57 @@ Resources:
...
+      MemorySize: 512
+      Layers:
+        - !Ref AWSSDKLayer
+        - !Ref FPLayer
+      Policies:
+        - AmazonDynamoDBFullAccess
+        - AmazonECS_FullAccess
+        - AWSLambdaBasicExecutionRole
+        - AWSLambdaInvocation-DynamoDB
+      Environment:
```
Comment (matteofigus) [inline src/cfn/template.yaml]: Should we use the Globals section on the top to allow all the lambdas to get all the variables for consistency? I would prefer to have either globals or lambda-specific for consistency

File: src/cfn/template.yaml [hunk 3]
```diff
@@ -453,6 +435,57 @@ Resources:
...
+        Variables:
+          CLUSTER: !Ref ECSCluster
+          RETRY_THRESHOLD: !Ref RetryThreshold
+          SUBNETS: !Sub ${PrivateSubnet1},${PrivateSubnet2},${PrivateSubnet3}
+          TASKS_TABLE_NAME: !Ref TasksDynamoTable
+          TASK_NAME: transcriber
```
Comment (matteofigus) [inline src/cfn/template.yaml]: Can we get this as reference from the `TransciberTaskDefinition`?

File: src/cfn/template.yaml [hunk 4]
```diff
@@ -453,6 +435,57 @@ Resources:
...
+      Policies:
+        - AmazonDynamoDBFullAccess
+        - AmazonECS_FullAccess
+        - AWSLambdaBasicExecutionRole
+        - AWSLambdaInvocation-DynamoDB
+      Environment:
+        Variables:
+          CLUSTER: !Ref ECSCluster
+          RETRY_THRESHOLD: !Ref RetryThreshold
+          SUBNETS: !Sub ${PrivateSubnet1},${PrivateSubnet2},${PrivateSubnet3}
```
Comment (matteofigus) [inline src/cfn/template.yaml]: In the globals we are already setting a `VPC_ID` which I think we are not going to use as we are just sending the subnets directly. Can we therefore remove `VPC_ID` if not needed?

File: src/cfn/template.yaml (hunk 1 above)
Comment (HieronymusLex) [reply src/cfn/template.yaml]: yeah I just took whatever the existing deployed timeout was but thinking about it you're right - we should at least have the default stop task timeout as a minimum which is 30 secs. so maybe 1 min to be safe?

File: src/cfn/template.yaml (hunk 3 above)
Comment (HieronymusLex) [reply src/cfn/template.yaml]: unfortunately not - AWS::ECS::TaskDefinition returns the ARN. but this is hardcoded in the template a bit too much for my liking - perhaps we could parameterise it then default to `transcriber`?

File: src/cfn/template.yaml (hunk 4 above)
Comment (HieronymusLex) [reply src/cfn/template.yaml]: yep good call

File: src/cfn/template.yaml (hunk 2 above)
Comment (HieronymusLex) [reply src/cfn/template.yaml]: maybe one for @svozza to confirm but I don't think any of the env vars are shared between the 2 so I'd prefer lambda-specific?

Comment (matteofigus) [reply src/cfn/template.yaml]: I think I'm ok with lambda specific, if you don't mind, would you move the current globals to the setup lambda?

File: src/cfn/template.yaml (hunk 3 above)
Comment (matteofigus) [reply src/cfn/template.yaml]: Yeah I think that would be a good idea

File: src/cfn/template.yaml (hunk 1 above)
Comment (matteofigus) [reply src/cfn/template.yaml]: yeah, 60 sounds like a good amount

**Pull request merged**